app/controllers/television_shows_controller.rb

```diff
@@ -1,9 +1,10 @@
 class TelevisionShowsController < ApplicationController
   def create
-    television_show = TelevisionShow.new(television_show_params)
+    television_show = TelevisionShow.find_or_create_by(television_show_params)
 
-    television_show.save
-    current_user.recommend(television_show)
+    unless current_user.recommend(television_show)
+      flash[:error] = t(".cannot_recommend_twice")
```

Comment:
Small point, but maybe instead of saying "hey you can't do that!" phrase it as "Nice Rec! But you already did this one," or some such.